### DISCUSSION TITLE
Picard iterations

### DISCUSSION THREAD

**Discussion by gabburgio** [idaholab/moose#27318] [Q&A General] [answered]
Hello everyone,

I would like to ask how one should go about using fixed-point iterations in a MOOSE nonlinear solve.
Specifically, assuming there are two (nonlinear) Kernels A(u) and B(u), is there some way to assign to one the role of the "source" in the fixed point scheme, i.e. if the equation is:
A(u) - B(u) = 0

Can one get MOOSE to solve:
A(u^{k+1}) = B(u^{k})
(with some other nonlinear solver)

This is mostly with the goal of implementing the algorithm at page 5 of https://doi.org/10.1016/j.anucene.2017.09.027, but also as a general question.

Thanks in advance for the help
Gabriele

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can set up fixed point iterations using the MultiApp system. 
https://mooseframework.inl.gov/syntax/MultiApps/index.html
There is a tutorial here
https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial02_multiapps/presentation/index.html#/

You set up two inputs for both equations and use the fixed point parameters in the Executioner of the main app

In the near future we will be able to set up fixed point iterations using multi-system. There is a test executioner that can be modified to do this for steady-state solves if you feel confident with coding

Guillaume
